metabase/metabase
# Define public metabase theme and Mantine transformer for embedding SDK

Closes https://github.com/metabase/metabase/issues/42201

### Description

Defines a standalone `MetabaseTheme` theme object. Modifies the `MetabaseProvider` to transforms the input `MetabaseTheme` theme object into a `MantineThemeOverride`, for consumption by the `ThemeProvider`.

This PR also moves the `font` configuration in `SDKConfig` to `fontFamily` theme option for consistency.

### How to verify

- Navigate to `basic-theming-public-theme-type` branch on the demo app.
- Navigate to the `AppProvider` component, verify the new theme configuration are in use.
- Change the color schemes or font family to make it easy to test. See the demo below.

### Demo

With the following theme option applied, notice that the loading spinner is now lime green (`brand`), and the chart text is yellow (`text-dark`).

```ts
const theme: MetabaseTheme = {
  fontFamily: "Lato",
  fontSize: "1rem",
  lineHeight: "1.1rem",
  colors: {
    brand: "lime",
    "text-dark": "yellow",
    "text-light": "green",
  },
}
```

https://github.com/metabase/metabase/assets/4714175/67743c9d-2f75-4dbd-b555-59dfc26c4531

### Checklist
